src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -215,6 +215,23 @@ public function get_items( $request ) {
 	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
 	 */
 	public function get_item_permissions_check( $request ) {
+		$template = $this->get_block_template( $request['id'] );
+		if ( is_wp_error( $template ) ) {
+			return $template;
+		}
+
+		if ( ! $template->wp_id ) {
+			return $this->permissions_check( $request );
+		}
+
+		if ( current_user_can( 'read_post', $template->wp_id ) ) {
+			return true;
+		}
+
+		if ( 'edit' === $request['context'] && current_user_can( 'edit_post', $template->wp_id ) ) {
+			return true;
+		}
+
 		return $this->permissions_check( $request );
```

Comment:
```suggestion
		return false;
```

I think this should return false, if someone has `edit_theme_options` and reaches this point, it's likely they've been blocked from editing the template post type by a theme or plugin developer using `map_meta_cap`.

This will prevent devs from doing this.

:1234: This applies to the other updated permission checks for various requests.

Reply:
This is retain backwards compat. This endpoint was added in 5.8. I don't want to change this to much.

Reply:
I agree with Peter. Otherwise we're creating incorrect expectations for developers that rejecting a user from being able to edit the particular post will actually prevent them from doing that.

Since these caps all map to `edit_theme_options` in the post type, I don't think a break here is likely either.

Reply:
I don't think that either of you understand how template / template parts work. Not all template map to a post object. Template parts can and added load via PHP. 
![Screenshot 2021-12-15 at 17 44 15](https://user-images.githubusercontent.com/237508/146237987-19430baa-ab0f-4fca-845e-ef65e6903f0e.png)

Template part with a wp_id, do not map to a post and do not need to respect the capabilities.